inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_gather_nd.cpp

```diff
@@ -0,0 +1,103 @@
+                     target, ngraph::opset5::GatherND::type_info);
+
+    auto shapeToConstant = [&gatherND](const ngraph::Output<ngraph::Node>& output,
+                                       const ngraph::element::Type& elemType) -> std::shared_ptr<ngraph::opset5::Constant> {
+        VPU_THROW_UNLESS(output.get_partial_shape().is_static(),
+                         "dynamicToStaticShapeGatherND transformation for {} of type {} expects static shape on inputs without DSR",
+                         gatherND->get_friendly_name(), gatherND->get_type_info());
+        return ngraph::opset5::Constant::create(elemType, {output.get_shape().size()}, output.get_shape());
+    };
+
+    const auto dataDSR = ngraph::as_type_ptr<ngraph::vpu::op::DynamicShapeResolver>(gatherND->input_value(0).get_node_shared_ptr());
+    const auto indicesDSR = ngraph::as_type_ptr<ngraph::vpu::op::DynamicShapeResolver>(gatherND->input_value(1).get_node_shared_ptr());
+
+    VPU_THROW_UNLESS(dataDSR || indicesDSR, "dynamicToStaticShapeGatherND transformation for {} of type {} expects at least one DSR as input",
+                     gatherND->get_friendly_name(), gatherND->get_type_info());
+    if (dataDSR && indicesDSR) {
+        VPU_THROW_UNLESS(dataDSR->get_input_element_type(1) == indicesDSR->get_input_element_type(1),
+                         "dynamicToStaticShapeGatherND transformation for {} of type {} expects equal shapes data types, actual {} vs {}",
+                         gatherND->get_friendly_name(), gatherND->get_type_info(),
+                         dataDSR->get_input_element_type(1), indicesDSR->get_input_element_type(1));
+    }
+    const auto shapeElementType = indicesDSR ? indicesDSR->get_input_element_type(1) : dataDSR->get_input_element_type(1);
+
+    const auto dataShape = dataDSR ? dataDSR->input_value(1) : shapeToConstant(gatherND->input_value(0), shapeElementType);
+    const auto indicesShape = indicesDSR ? indicesDSR->input_value(1) : shapeToConstant(gatherND->input_value(1), shapeElementType);
+
+    const auto dataShapeRank = static_cast<size_t>(dataShape.get_shape()[0]);
+    const auto indicesShapeRank = static_cast<size_t>(indicesShape.get_shape()[0]);
+
+    int64_t batchDims = gatherND->get_batch_dims();
+    VPU_THROW_UNLESS(batchDims >= 0 && batchDims < std::min(dataShapeRank, indicesShapeRank),
+                     "dynamicToStaticShapeGatherND: node {} has unsupported batch_dims which is expected to be"
+                     " in [0; min({}, {})), but {} was provided", gatherND->get_friendly_name(), dataShapeRank, indicesShapeRank, batchDims);
+
+    VPU_THROW_UNLESS(indicesShapeRank - batchDims >= 2,
+                     "dynamicToStaticShapeGatherND: node {} should have at least two non-batch dimension, {} provided",
+                     gatherND->get_friendly_name(), indicesShapeRank - batchDims);
```

Comment:
Is it constrain of VPU implementation?

Reply:
Actually it was the constrain of the DTS itself, but now it's not necessary, as I've remade it.